openff/toolkit/topology/topology.py

```diff
@@ -2494,6 +2494,11 @@ def atom(self, atom_topology_index):
                 return molecule.atom(atom_molecule_index)
             this_molecule_start_index += molecule.n_atoms
 
+        raise AtomNotInTopologyError(
+            f"No atom with index {atom_topology_index} exists in this topology, "
+            f"which contains {self.n_atoms} atoms."
+        )
```

Comment:
This function's logic has an expression after the `return` statement, which confused the type checker. I threw this in not because it should ever be hit, but maybe it's possible.